hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/validators/AddressBookValidator.java

```diff
@@ -167,4 +173,18 @@ private void validateEndpoint(@NonNull final ServiceEndpoint endpoint, @NonNull
         validateFalse(endpoint.domainName().trim().length() > nodesConfig.maxFqdnSize(), FQDN_SIZE_TOO_LARGE);
         validateFalse(addressLen != 0 && addressLen != 4, INVALID_IPV4_ADDRESS);
     }
+
+    /**
+     *  Validate the Bytes is a real X509Certificate bytes.
+     * @param certBytes the Bytes to validate
+     * @throws PreCheckException
+     */
+    public static void validateX509Certificate(@NonNull Bytes certBytes) throws PreCheckException {
+        try {
+            final var cert = (X509Certificate) CertificateFactory.getInstance("X.509")
+                    .generateCertificate(new ByteArrayInputStream(certBytes.toByteArray()));
```

Comment:
This parses the certificate, but do we want to check validity and possibly verify the signatures up the cert path to the CA?

Reply:
We only check if it's X509 Certificate bytes here.